### PR TITLE
Dev/master ecbuild sync

### PR DESCRIPTION
**Pull request by aerorahul**
syncing NOAA-GFDL/mom6::dev/master with JCSDA/mom6::dev/master and then updating JCSDA/mom6::dev/master-ecbuild

There are extensive changes in MOM6.
There were no conflicts (amazingly).  We should cron this repo to automagically sync, if we wish to remain in sync with them.  Sooner (or later) we will deal directly with a coupled model (either UFS or GEOS), and this may become moot, but until then, this needs doing every so often.

Suggestions?

### CODE REVIEW

**Comment by travissluka** [comment]
I'm getting an error when I try to build
```
CMake Error at /usr/local/share/ecbuild/cmake/ecbuild_add_library.cmake:270 (add_library):
  Cannot find source file:

    src/user/SCM_CVmix_tests.F90
```

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -168,15 +169,16 @@ src/user/dyed_channel_initialization.F90
...
+src/user/Idealized_Hurricane.F90
 src/user/Kelvin_initialization.F90
 src/user/lock_exchange_initialization.F90
 src/user/MOM_controlled_forcing.F90
 src/user/MOM_wave_interface.F90
 src/user/Neverland_initialization.F90
 src/user/Phillips_initialization.F90
+src/user/RGC_initialization.F90
 src/user/Rossby_front_2d_initialization.F90
 src/user/SCM_CVmix_tests.F90
```
Comment (travissluka) [inline src/CMakeLists.txt]:
```suggestion
src/user/SCM_CVMix_tests.F90
```

**Review by travissluka — CHANGES_REQUESTED** [review]
Some of the interfaces have changed slightly. We'll have to prepare changes in SOCA first before merging this in.

**Comment by aerorahul** [comment]
@travissluka I agree we should update SOCA interfaces before merging this branch.
I can work on exploring which interfaces need to be updated and how many they are.

**Comment by guillaumevernieres** [comment]
Sounds good @aerorahul , thanks for doing that work.

**Comment by aerorahul** [comment]
@travissluka @guillaumevernieres 
See branch `soca::feature/mom6-change` that incorporates the changes in MOM6 into the SOCA interface.

**Review by travissluka — APPROVED** [review]

**Pull request merged**